homeassistant/components/firmata/__init__.py

```diff
@@ -0,0 +1,142 @@
+    DOMAIN,
+    PIN_MODE_INPUT,
+    PIN_MODE_OUTPUT,
+    PIN_MODE_PULLUP,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_CONFIGS = "board_configs"
+
+ANALOG_PIN_SCHEMA = vol.All(cv.string, vol.Match(r"^A[0-9]+$"))
+
+SWITCH_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_PIN): vol.Any(cv.positive_int, ANALOG_PIN_SCHEMA),
+        # will be analog mode in future too
+        vol.Required(CONF_PIN_MODE): PIN_MODE_OUTPUT,
+        vol.Optional(CONF_INITIAL_STATE, default=False): cv.boolean,
+        vol.Optional(CONF_NEGATE_STATE, default=False): cv.boolean,
+    },
+    required=True,
+    extra=vol.ALLOW_EXTRA,
+)
+
+BINARY_SENSOR_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_NAME): cv.string,
+        vol.Required(CONF_PIN): vol.Any(cv.positive_int, ANALOG_PIN_SCHEMA),
+        # will be analog mode in future too
+        vol.Required(CONF_PIN_MODE): vol.Any(PIN_MODE_INPUT, PIN_MODE_PULLUP),
+        vol.Optional(CONF_NEGATE_STATE, default=False): cv.boolean,
+    },
+    required=True,
+    extra=vol.ALLOW_EXTRA,
+)
+
+BOARD_CONFIG_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_SERIAL_PORT): cv.string,
+        vol.Optional(CONF_SERIAL_BAUD_RATE): cv.positive_int,
+        vol.Optional(CONF_ARDUINO_INSTANCE_ID): cv.positive_int,
+        vol.Optional(CONF_ARDUINO_WAIT): cv.positive_int,
+        vol.Optional(CONF_SLEEP_TUNE): vol.All(
+            vol.Coerce(float), vol.Range(min=0.0001)
+        ),
+        vol.Optional(CONF_SAMPLING_INTERVAL): cv.positive_int,
+        vol.Optional(CONF_SWITCHES): [SWITCH_SCHEMA],
+        vol.Optional(CONF_BINARY_SENSORS): [BINARY_SENSOR_SCHEMA],
+    },
+    required=True,
+    extra=vol.ALLOW_EXTRA,
+)
+
+CONFIG_SCHEMA = vol.Schema(
+    {DOMAIN: vol.All(cv.ensure_list, [BOARD_CONFIG_SCHEMA])}, extra=vol.ALLOW_EXTRA
+)
+
+
+async def async_setup(hass, config):
+    """Set up the Firmata domain."""
+    # Delete all current entries and re-add them to update them
+    if hass.config_entries.async_entries(DOMAIN) and DOMAIN in config:
+        for entry in hass.config_entries.async_entries(DOMAIN):
+            await hass.config_entries.async_remove(entry.entry_id)
+    if DOMAIN in config:
+        for board in config[DOMAIN]:
+            firmata_config = deepcopy(board)
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": config_entries.SOURCE_IMPORT},
+                    data=firmata_config,
+                )
+            )
+    return True
+
+
+async def async_setup_entry(hass, config_entry):
+    """Set up a Firmata board for a config entry."""
+    if DOMAIN not in hass.data:
+        hass.data[DOMAIN] = {}
+
+    _LOGGER.debug(
+        "Setting up Firmata id %s, config %s", config_entry.entry_id, config_entry.data
+    )
+    if not config_entry.options:
+        hass.config_entries.async_update_entry(config_entry)
```

Comment:
You're not passing any info to update the config entry?

Reply:
What kind of info should I be passing here?

Reply:
What ever you want to update. If you don't want to update the config entry, don't call update.